tests/testthat/setup-workflow-ref.R

```diff
@@ -170,3 +170,15 @@ cleanup <- function() {
   suppressSpecificWarning(rm(log_df, pos = parent.frame()),.regexpr = "object.+not found")
 }
 
+#' Temporarily perturb a file
+#'
+#' Appends a line to the end of a text file, and flexibly revert the change.
+#'
+#' @param path string giving the file path
+#' @inheritParams withr::defer
+perturb_file <- function(path, envir = parent.frame()) {
```

Comment:
this function is sexy

Reply:
Thanks, my first attempt at implementing a ["local" helper](https://testthat.r-lib.org/articles/test-fixtures.html#-local-helpers).